### DISCUSSION TITLE
Getting NextRouter was not mounted when running next js in storybook.

### DISCUSSION THREAD

**Discussion by tmprakash3** [storybookjs/storybook#25310] [Help] [answered]
### Summary

<img width="1269" alt="image" src="https://github.com/storybookjs/storybook/assets/15902585/04357e91-14ab-4597-b8fb-20f8fa76f750">
Hi,
I'm using Nextjs 14.0.3 and Storybook of the 6.3.7 version. while upgrading Nexjs useRouter from next/router to next/navigation from 12 to 14 getting this error on the storybook and added all the necessary changes in the main.ts and preview.ts files, but still getting the same error, and we are using the mono repo repository. If i commented out router reference in the components it is loading fine.

### Additional information

main.ts
import { dirname, join } from 'path'
module.exports = {
  stories: ['../src/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  typescript: { reactDocgen: 'react-docgen' },

  addons: [
    {
      name: '@storybook/addon-docs',
      options: {
        configureJSX: true,
        babelOptions: {},
        sourceLoaderOptions: null,
      },
    },
    "storybook-addon-next-router",
    // getAbsolutePath("@storybook/addon-links"),
    // getAbsolutePath("@storybook/addon-essentials"),
    // getAbsolutePath("storybook-addon-next-router"),
  ],

  framework: {
    // name: '@storybook/react-webpack5', // Remove this
    //name: getAbsolutePath('@storybook/nextjs'), // Add this
    name: '@storybook/nextjs',
    options: {},
  },
  webpackFinal: (config: any) => {
    config.node = { fs: 'empty' }
    return config
  },

  docs: {
    autodocs: true,
  },
}

// function getAbsolutePath(value: any) {
//   return dirname(require.resolve(join(value, 'package.json')))
// }

**Comment by valentinpalkovic** [comment, marked as answer]
Storybook 6 is not maintained anymore. We have added exceptional Next.js support in Stoybook 7. Please upgrade your Storybook to the latest version.

**Reply by tmprakash3** [reply]
Thanks @valentinpalkovic  after upgrading @storybook/nextjs to 7, it's working now.

**Reply by tmprakash3** [reply]
we are still getting below error in some of the components. The configuration is same as above.
<img width="1273" alt="image" src="https://github.com/storybookjs/storybook/assets/15902585/f3e09024-7012-4835-a335-6877b39cc06d">

**Reply by valentinpalkovic** [reply]
Can you take a look at https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/README.md#nextjs-navigation to be sure that you have set all the necessary options?
